Add json-navigator

DamienCassou opened the pull request:
### Brief summary of what the package does
View and navigate JSON structures

### Direct link to the package repository

https://github.com/DamienCassou/json-navigator

### Your association with the package
package maintainer

### Relevant communications with the upstream package maintainer

**None needed**

### Checklist

Please confirm with `x`:

- [x] I've read [CONTRIBUTING.md](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.md)
- [x] I've used [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] `M-x checkdoc` is happy with my docstrings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.md](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.md)

purcell commented:
Thanks!

- There's no dependency on `hierarchy`, so you must declare one.
- Consider adding a command `json-navigator-region`, which allows browsing a selected portion of JSON text in a buffer. You could make its region default to the full contents of the current buffer.

DamienCassou commented:
Good ideas, thanks. I fixed both.

Note: I knew about `"There's no dependency on hierarchy"` but it took me some time to make package-lint happy about it (issue https://github.com/purcell/package-lint/issues/55). I fixed it by redoing what you do in https://github.com/purcell/package-lint/blob/master/run-tests.sh.

purcell commented:
Cool, merged. And now you can remove `(depends-on "hierarchy" "0.5.0")` from `Cask`.
